server/src/main/java/org/elasticsearch/search/sort/SortValue.java

```diff
@@ -338,4 +346,69 @@ public Number numberValue() {
             return Double.NaN;
         }
     }
+
+    private static class EmptySortValue extends SortValue {
```

Comment:
Since the whole thing is immutable you can just have one static instance I think.